nodejs/src/lib.ts

```diff
@@ -176,6 +215,38 @@ export function sign(args: SignArguments) {
   });
 }
 
+/** Sign/verification error
+ * SignatureError: SignatureError,
+ * Produce a JWS `Tl-Signature` v2 header value.
+ * @param {Object} args - Arguments.
+ * @param {string} args.signCallback - Callback to sign using a KMS/HSM.
+ * @param {string} args.kid - Private key kid.
+ * @param {string} [args.method="POST"] - Request method, e.g. "POST".
+ * @param {string} args.path - Request path, e.g. "/payouts".
+ * @param {string} [args.body=""] - Request body.
+ * @param {Object} [args.headers={}] - Request headers to be signed.
+ * Warning: Only a single value per header name is supported.
+ * @returns {Promise<string>} Tl-Signature header value.
+ * @throws {SignatureError} Will throw if signing fails.
+ */
+export function signWithCallback(args: SignWithCallbackArguments): Promise<string> {
```

Comment:
[nit] If we are exporting a function we should also export the type of the args. This is so our users can also use the args type in their code

Reply:
I did wonder that, but saw we don't for the existing `sign` function, so was being blindly consistent. I see we do export all the types used by the exported `verify` functions. I'll update both `sign` functions...

Reply:
It seems we are generally inconsistent with what we export, we export [VerifyParameters](https://github.com/TrueLayer/truelayer-signing/blob/75fee2e110d7a780ae87473ba32684995228d714/nodejs/src/lib.ts#L296) which is used in the [verify function  ](https://github.com/TrueLayer/truelayer-signing/blob/75fee2e110d7a780ae87473ba32684995228d714/nodejs/src/lib.ts#L315) but we don't export the types for any for the signing functions.

Reply:
Actually that has made me notice we have multiple overloads of `verify` - perhaps I should rename`signWithCallback` to simply be an overload of `sign`?

Reply:
I don't think we should overload, I think generally in javascript it's idiomatic for the callback to a separate argument, so I would keep it seperate and write it like this: 
```
export type signCallback = (payload: string) => Promise<string>;

export function signWithCallback(args: SignArgumentsCommon, callback: ): Promise<string> {
```

Reply:
Isn't that more for callbacks that are continuations?
Perhaps me calling this a callback is confusing - it more like dependency injection, in that the function passed is used by this function to do the actual signing.

Reply:
Updated to remove the callback terminology